store/persist.ts

```diff
@@ -17,8 +17,51 @@ export interface PersistProps<S extends AnyState> {
   key: string;
   reconciler: (original: S, rehydrated: Partial<S>) => S;
   rehydrate: () => Operation<Result<unknown>>;
+  transform?: TransformFunctions<S>;
 }
 
+interface TransformFunctions<S extends AnyState> {
+  in(s: Partial<S>): Operation<Partial<S>>;
```

Comment:
I wonder, do we want to be able to perform IO in this layer of our store?  It kind of feels like that could cause serious issues since this function is going to be hit everytime there's a state update.

Reply:
The inbound transformer (in) processes the state before it is persisted. It does not perform any IO (input/output) operations, and is purely focused on transforming the state within the application.

Similarly, the outbound transformer (out) processes the state after it is re-hydrated from storage. This function is also purely functional and does not interact with any external systems.

The naming of the functions in and out may be misleading, as they could imply external IO operations are involved. However, both functions are strictly concerned with internal state transformations and do not handle any external side effects.

Reply:
However, since users are able to provide custom transformer functions, it is possible for IO operations or side effects to be introduced if the user chooses to include them. 
These kinds of issues were not addressed in the submitted code. 
I believe we need to explore ways to enforce purity in the transformers, ensuring they remain free of side effects.

Reply:
Yeah, I would make these normal functions and not ones that return `Operation`